### PR TITLE
Update build.yml

### PR DESCRIPTION
**Pull request by VDubber**
Update to reference `main` instead of `master`

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,70 +1,51 @@
...
+  BuildRelease:
     needs: test
     runs-on: ubuntu-latest
-    if: github.ref == 'refs/heads/master'
-    strategy:
-      fail-fast: false
-      matrix:
-        node: [14]
-
+    if: github.ref == 'refs/heads/main'
```
Comment (VDubber) [inline .github/workflows/build.yml:35]: Root fix @eXtremeX

**Review by eXtremeX — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (eXtremeX) [reply .github/workflows/build.yml:35]: Awesome, let's see if that was it!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,70 +1,51 @@
...
-          fi
-      - name: Publish
-        if: env.publish == 'true'
-        env:
-          NPM_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
-        run: |
-          echo "//registry.npmjs.org/:_authToken=${NPM_AUTH_TOKEN}" > .npmrc
-          yarn
-          npm publish
+          token: ${{ secrets.AUTO_GITHUB_PAT_TOKEN }}
```
Comment (adam-in-ict) [inline .github/workflows/build.yml:46]: Have we gotten someone with permissions to update the settings for this repo to be able to use `auto`?

Comment (eXtremeX) [reply .github/workflows/build.yml:46]: There's this ticket: https://jupiterone.atlassian.net/browse/DESK-2166, that might be it.

Comment (eXtremeX) [reply .github/workflows/build.yml:46]: But I'm not sure if something else other than secrets need to be updated for auto.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -1,70 +1,51 @@
 name: Build
-on: [push, pull_request]
+on:
+  pull_request:
```
Comment (mdaum) [inline .github/workflows/build.yml:3]: Do you want build to run on all pull requests? Or just ones to `main` branch. If you want the latter change this to:
```
pull_request:
  branches:
    - main
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -1,70 +1,51 @@
...
-        if: env.publish == 'true'
-        env:
-          NPM_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
-        run: |
-          echo "//registry.npmjs.org/:_authToken=${NPM_AUTH_TOKEN}" > .npmrc
-          yarn
-          npm publish
+          token: ${{ secrets.AUTO_GITHUB_PAT_TOKEN }}
+      - name: Build and Release
+        uses: jupiterone/action-npm-build-release@v1
```
Comment (mdaum) [inline .github/workflows/build.yml:48]: have not used this workflow before. Have we used it in other integrations? If not I'll look over the workflow and give my 2 cents.

Comment (VDubber) [reply .github/workflows/build.yml:48]: This uses auto shipit. We use it in other graph projects include graph-github

**Review by adam-in-ict — APPROVED** [review]

**Pull request merged**